File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsPlatformSettingsAdmin.swift
import Foundation

/// Secret-free subset of `GET /api/v1/settings/platform` used by mobile admin.
/// Unknown fields (including masked secret fields) are intentionally not decoded.
struct PlatformSettingsSnapshot: Decodable, Equatable {
    var ffFeedback: Bool
    var ffPublicCatalog: Bool
    var ffCourseMarketplace: Bool
    var ffLearningPaths: Bool
    var ffPeerReview: Bool
    var ffCompletionCredentials: Bool
    var ffPersistentTutor: Bool
    var ffAiStudyBuddy: Bool
    var ffClassroomSignals: Bool
    var ffBroadcasts: Bool
    var ffCalendarFeeds: Bool
    var learnerProfileEnabled: Bool

    var samlSsoEnabled: Bool
    var samlPublicBaseUrl: String
    var samlSpEntityId: String
    var mfaEnabled: Bool
    var mfaEnforcement: String
    var smtpHost: String
    var smtpPort: Int
    var smtpFrom: String
}

/// Effective states from the public runtime feature endpoint. Config fields are
/// deliberately absent because this payload is used only to override flag state.
struct PlatformFeatureStates: Decodable {
    var ffFeedback: Bool?
    var ffPublicCatalog: Bool?
    var ffCourseMarketplace: Bool?
    var ffLearningPaths: Bool?
    var ffPeerReview: Bool?
    var ffCompletionCredentials: Bool?
    var ffPersistentTutor: Bool?
    var ffAiStudyBuddy: Bool?
    var ffClassroomSignals: Bool?
    var ffBroadcasts: Bool?
    var ffCalendarFeeds: Bool?
    var learnerProfileEnabled: Bool?
}

struct PlatformFeatureDefinition: Identifiable, Equatable {
    let key: String
    let labelKey: String
    let descriptionKey: String

    var id: String { key }
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsPlatformSettingsAdmin.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

/** Secret-free subset of the platform settings response used by mobile admin. */
@Serializable
data class PlatformSettingsSnapshot(
    val ffFeedback: Boolean = false,
    val ffPublicCatalog: Boolean = false,
    val ffCourseMarketplace: Boolean = false,
    val ffLearningPaths: Boolean = false,
    val ffPeerReview: Boolean = false,
    val ffCompletionCredentials: Boolean = false,
    val ffPersistentTutor: Boolean = false,
    val ffAiStudyBuddy: Boolean = false,
    val ffClassroomSignals: Boolean = false,
    val ffBroadcasts: Boolean = false,
    val ffCalendarFeeds: Boolean = false,
    val learnerProfileEnabled: Boolean = false,
    val samlSsoEnabled: Boolean = false,
    val samlPublicBaseUrl: String = "",
    val samlSpEntityId: String = "",
    val mfaEnabled: Boolean = false,
    val mfaEnforcement: String = "none",
    val smtpHost: String = "",
    val smtpPort: Int = 587,
    val smtpFrom: String = "",
)

@Serializable
data class PlatformFeatureStates(
    val ffFeedback: Boolean = false,
    val ffPublicCatalog: Boolean = false,
    val ffCourseMarketplace: Boolean = false,
    val ffLearningPaths: Boolean = false,
    val ffPeerReview: Boolean = false,
    val ffCompletionCredentials: Boolean = false,
    val ffPersistentTutor: Boolean = false,
    val ffAiStudyBuddy: Boolean = false,
    val ffClassroomSignals: Boolean = false,
    val ffBroadcasts: Boolean = false,
    val ffCalendarFeeds: Boolean = false,
    val learnerProfileEnabled: Boolean = false,
)

data class PlatformFeatureDefinition(
    val key: String,
    val labelResName: String,
    val descriptionResName: String,
)
